lib/travis/build/script/node_js.rb

```diff
@@ -208,6 +210,18 @@ def node_0_9?
           def use_npm_cache?
             Array(config[:cache]).include?('npm')
           end
+        
+          def has_package_lock_file?
+            sh.if "-f package-lock.json"
+          end
+        
+          def has_shrinkwrap_file?
+            sh.if "-f npm-shrinkwrap.json"
```

Comment:
Is this valid ruby?

Reply:
Ruby only generates a bash script. So I don't think this will work. 

You could instead make nest the `sh.if`s here: https://github.com/btkostner/travis-build/blob/d6d0d6ec2b22cee1a6473369a0e701923775187d/lib/travis/build/script/node_js.rb#L239.

So maybe somethin like this would work:
``` ruby 
              sh.if "$(vers2int `npm -v`) -gt $(vers2int #{NPM_CI_CMD_VERSION})" do
                     sh.if "-f npm-shrinkwrap.json -o -f package-lock.json" do
                         sh.cmd "npm ci #{args}", retry: true
                     else
                         sh.cmd "npm install #{args}", retry: true
                    end
              end
```

Alternatively making the first `sh.if` more complex would probably be nicer.